### PR TITLE
post order provider

### LINKED ISSUE #53: Post/provider/orders
#9 #18
### Back-end Routes Template
----
- [ ] **Route Name**

* full-route ` /provider/orders/` 
* method : `Post ` 
---- 
#### Acceptance Criteria
- [ ] **Validation schema**
 {      
arrive_time: time().required(),
}
---- 
- [ ]  **Request body** 
 ```js 
  req.body =  {
       arrive_time: 06:25  pm
   }
```
----
- [ ] **Response**
- Success

```js
{ statusCode: 200 ,message  : "order request accepted successfully "}
```
- Failed
```js
{statusCode:500 , message:"something wrong "}
```

### PR DESCRIPTION
**Pull request by zeinjendeya**
- created a 3 query:
    - get Order By ID: to check the status
    - change State Order Request: to change the state in the order-request table from pending to accepted
    - accept orders: to insert the new order into the orders table with the arrival time and state
- created a controller: to use all three queries and check state then insert then send a request with 201 status code and a message that the request was accepted successfully. 

#53

### CODE REVIEW

File: server/controller/provider/orderAccept.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  try {
+    const {
+      rows: [{ id, state }],
+    } = await getOrderByUserID(userID, orderID);
+
+    if (state === 'pending') {
+      await updateStateOrderRequest(id);
+    } else {
+      throw boomify(409, 'order already accepted');
+    }
```
Comment (mariamisa) [inline server/controller/provider/orderAccept.js]: you can directly update the request without getting the data

File: server/database/queries/provider/changeStateOrderRequest.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const connection = require('../../config/connection');
+
+const updateStateOrderRequest = (orderID) => {
+  const sql = {
+    text: "UPDATE ordes_request SET state='accepted' WHERE id=$1'",
```
Comment (mariamisa) [inline server/database/queries/provider/changeStateOrderRequest.js]: you need to check if the owner of this order is the logged provider

File: server/database/queries/provider/getOrderProvider.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+const getOrderByUserID = (userID, orderID) => {
+  const sql = {
+    text: 'SELECT * FROM ordes_request WHERE user_id=$1 AND id=$2',
+    values: [userID, orderID],
+  };
+
+  return connection.query(sql);
+};
+
+module.exports = getOrderByUserID;
```
Comment (mariamisa) [inline server/database/queries/provider/getOrderProvider.js]: no need to get orderRequest

File: server/router/user.js [hunk 4]
```diff
@@ -1,7 +1,10 @@
 const router = require('express').Router();
 
 const { isAuthController } = require('../controller/user');
+const { orderAcceptValidation } = require('../middleware/validation');
+const { acceptOrderController } = require('../controller');
 
 router.get('/is-auth', isAuthController);
+router.post('/provider/orders', orderAcceptValidation, acceptOrderController);
```
Comment (mariamisa) [inline server/router/user.js]: this route relates to the provider file and needs to providerProtectedMIddelware the user shouldn't can access these endpoints

**Review by mariamisa — CHANGES_REQUESTED** [review]

**Review by mAbadsa — DISMISSED** [review]

**Review by mariamisa — APPROVED** [review]

**Pull request merged**